modules/openapi-generator/src/main/resources/python/python-experimental/api_client.mustache

```diff
@@ -542,3 +569,180 @@ class ApiClient(object):
+            digest = SHA256.new()
+            prefix = "SHA-256="
+        else:
+            raise Exception(
+                "Unsupported signing algorithm: {0}".format(self.configuration.signing_scheme))
+        digest.update(data)
+        return digest, prefix
+
+    def sign_digest(self, digest):
+        """
+        Signs a message digest with a private key specified in the configuration.
+
+        :param digest: A hashing object that contains the cryptographic digest of the HTTP request.
+        :return: A base-64 string representing the cryptographic signature of the input digest.
+        """
+        self.configuration.load_private_key()
+        privkey = self.configuration.private_key
+        if isinstance(privkey, RSA.RsaKey):
+            if self.configuration.signing_algorithm == 'PSS':
+                # RSASSA-PSS in Section 8.1 of RFC8017.
+                signature = pss.new(privkey).sign(digest)
+            elif self.configuration.signing_algorithm == 'PKCS1v15':
+                # RSASSA-PKCS1-v1_5 in Section 8.2 of RFC8017.
+                signature = PKCS1_v1_5.new(privkey).sign(digest)
+            else:
+                raise Exception("Unsupported signature algorithm: {0}".format(self.configuration.signing_algorithm))
+        elif isinstance(privkey, ECC.EccKey):
+            if self.configuration.signing_algorithm in ECDSA_KEY_SIGNING_ALGORITHMS:
+                signature = DSS.new(privkey, self.configuration.signing_algorithm).sign(digest)
+            else:
+                raise Exception("Unsupported signature algorithm: {0}".format(self.configuration.signing_algorithm))
+        else:
+            raise Exception("Unsupported private key: {0}".format(type(privkey)))
+        return b64encode(signature)
+
+    def get_authorization_header(self, signed_headers, signed_msg, created_ts):
+        """
+        Calculates and returns the value of the 'Authorization' header when signing HTTP requests.
+        
+        :param signed_headers : A list of strings. Each value is the name of a HTTP header that
+           must be included in the HTTP signature calculation.
+        :param signed_msg: A base-64 encoded string representation of the signature.
+        :param created_ts: The string representation of the time when the signature was created,
+           as a UNIX timestamp value.
+        :return: The string value of the 'Authorization' header, representing the signature
+           of the HTTP request.
+        """
+
+        headers_value = ""
+        is_created_set = False
+        for i, key in enumerate(signed_headers)
+            key = key.lower()
+            headers_value += key
+            if i > 0:
+                headers_value += " "
+            if key == '(created)':
+                is_created_set = True
+
+        auth_str = "Signature keyId=\"{0}\",algorithm=\"{1}\","
+            .format(self.configuration.key_id, self.configuration.signing_scheme)
+        if is_created_set:
```

Comment:
how about changing this to `if created_ts is not None:`

Reply:
done